domain-model/src/main/java/uk/gov/hmcts/cmc/domain/constraints/ValidAdmissionConstraintValidator.java

```diff
@@ -11,34 +11,45 @@
 import static uk.gov.hmcts.cmc.domain.models.PaymentOption.FULL_BY_SPECIFIED_DATE;
 import static uk.gov.hmcts.cmc.domain.models.PaymentOption.IMMEDIATELY;
 import static uk.gov.hmcts.cmc.domain.models.PaymentOption.INSTALMENTS;
+import static uk.gov.hmcts.cmc.domain.models.response.ResponseType.PART_ADMISSION;
 
-public class ValidFullAdmissionConstraintValidator
-    implements ConstraintValidator<ValidFullAdmission, FullAdmissionResponse> {
+public class ValidAdmissionConstraintValidator
+    implements ConstraintValidator<ValidAdmission, AdmissionResponse> {
 
     public static class Fields {
         public static final String PAYMENT_DATE = "paymentDate";
         public static final String REPAYMENT_PLAN = "repaymentPlan";
+        public static final String PAYMENT_OPTION = "paymentOption";
     }
 
     @Override
-    public boolean isValid(FullAdmissionResponse fullAdmissionResponse, ConstraintValidatorContext context) {
-        if (fullAdmissionResponse == null) {
+    public boolean isValid(AdmissionResponse admissionResponse, ConstraintValidatorContext context) {
+        if (admissionResponse == null) {
             return true;
         }
 
-        switch (fullAdmissionResponse.getPaymentOption()) {
+        if (admissionResponse.getPaymentOption() == null) {
```

Comment:
This does not look correct. You can just have this block and rest should fall through the below switch statement.
```
if (admissionResponse.getResponseType() == PART_ADMISSION) {
                return true;
 } 
```